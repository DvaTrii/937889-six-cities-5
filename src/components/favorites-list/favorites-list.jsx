import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import FavoritesItem from "../favorites-item/favorites-item";
import {getFavoriteOffersCities, getFavoritesOffers} from "../../store/data/selectors";


const FavoritesList = ({offers, cities}) => {

  return (
    <div className="page__favorites-container container">
      <section className="favorites">
        <h1 className="favorites__title">Saved listing</h1>
        <ul className="favorites__list">
          {cities.map((it) => <FavoritesItem key={it} offers={offers.includes(it)}/>)}
        </ul>
      </section>
    </div>
  );
};

FavoritesList.propTypes = {
  offers: PropTypes.array.isRequired,
  cities: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  return ({
    offers: getFavoritesOffers(state),
    cities: getFavoriteOffersCities(state)
  });
};

export {FavoritesList};
export default connect(mapStateToProps)(FavoritesList);
