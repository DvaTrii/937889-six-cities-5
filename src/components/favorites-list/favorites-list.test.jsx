import React from "react";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {FavoritesList} from "./favorites-list";

const mockStore = configureStore([]);

it(`Should FavoritesList render properly`, () => {

  const store = mockStore({
    APP: {hoveredOfferId: -1},
  });
  const cities = [`Amsterdam`];

  const offers = [{
    id: 101,
    city: {
      name: `Amsterdam`,
      coordinates: { // был [33, 33]
        latitude: 52.38333,
        longitude: 4.9
      }
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`],
    isPremium: true,
    isBookmark: true,
    price: 100,
    title: `Wood and stone place`,
    type: `House`,
    rating: 5,
    description: `Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium`,
    bedroomsMax: 4,
    guestsMax: 5,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Ross`,
      isSuper: true,
    },
  }];

  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <FavoritesList
              offers={offers}
              cities={cities}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
