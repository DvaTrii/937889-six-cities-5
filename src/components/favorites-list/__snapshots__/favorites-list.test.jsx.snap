// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesList render properly 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites"
  >
    <h1
      className="favorites__title"
    >
      Saved listing
    </h1>
    <ul
      className="favorites__list"
    >
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="#"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="favorites__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="110"
                  src="img/apartment-02.jpg"
                  width="150"
                />
              </a>
            </div>
            <div
              className="favorites__card-info place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    100
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button
              place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/101"
                  onClick={[Function]}
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                House
              </p>
            </div>
          </article>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
